feat(search): Added search page - FRONT-2348

tibi2303 opened the pull request:
The multiselect will be added after the Multiselect PR will be accepted and merged.

papegaill commented on src/compositions/search/search.html.twig:
```diff
@@ -0,0 +1,62 @@
+  {% include '@oe-bcl/navbar/navbar.html.twig' with data.navbar only %}
+</header>
+<main>
+  <div class="container mt-5">
```
I think we need to discuss this at team level, for me we would need a general class like `.bcl-search` and not use utility classes for spacing and styles, this should be defined in a dedicated sass file. cc: @planctus @emeryro 
More classes should be added for the sub-element like title etc..  to be able to target and style them directly via sass..

papegaill commented on storybook/.storybook/preview-head.html:
```diff
@@ -51,6 +51,42 @@
+          offcanvasMobile.classList.add("offcanvas", "offcanvas-start");
+        } else {
+          offcanvasMobile.classList.remove("offcanvas", "offcanvas-start");
+          offcanvasMobile.style.visibility = "visible";
```
I think we should add `bsOffcanvas.hide();` here to ensure the offcanvas closed if resizing to desktop

papegaill replied:
I think we should rename variable and classes names to something related to `bcl-search` as a prefix

papegaill commented on storybook/.storybook/preview-head.html:
```diff
@@ -51,6 +51,42 @@
+          logWidth();
+        },
+        false
+      );
```
do we have a plan of how this will be shiped .. or do we create a new ticket for this?

papegaill reviewed (commented):
Nice work 👍 here are some remarks:

The close btn icon is missing on the offcanvas:
<img width="237" alt="Screenshot 2021-06-24 at 18 48 46" src="https://user-images.githubusercontent.com/1289480/123302444-02a3f700-d51d-11eb-8d65-0a66b79354b6.png">
Horizontal padding missing on the offcanvas for tablet format:
<img width="405" alt="Screenshot 2021-06-24 at 18 46 26" src="https://user-images.githubusercontent.com/1289480/123302448-033c8d80-d51d-11eb-8e8c-a46ad4ebecc8.png">
Filter option icon is not visible:
<img width="722" alt="Screenshot 2021-06-24 at 18 45 47" src="https://user-images.githubusercontent.com/1289480/123302453-03d52400-d51d-11eb-936d-5c37279f12d5.png">
Content is cut within the sidebar, due to `.offcanvas-body `class (this class needs to be toggle in js, I guess?)..
<img width="291" alt="Screenshot 2021-06-24 at 18 32 57" src="https://user-images.githubusercontent.com/1289480/123302457-046dba80-d51d-11eb-9c99-52c07e5c0741.png">

papegaill reviewed (commented):
seems to be a style issue on all stories when viewing in fullscreen via the top left icon cc:@planctus

tibi2303 commented on src/components/badge/badge.html.twig:
```diff
@@ -6,6 +6,7 @@
   - url (string) (default: '')
   - title (string) (default: label)
   - hidden_text (string) (default: '')
+  - dismissible (boolean) (default: false)
```
I added a control for dismiss in badge.

papegaill commented:
@tibi2303 FYI since lastest changes, it seems that the search preview is broken.

tibi2303 commented:
@papegaill should work now, i had updated the data accordingly but it had a conflict in badge, probably you will need to add the modifications in patterns after merge also, I added dismissible param.
